vernac/mltop.ml

```diff
@@ -334,23 +333,38 @@ let _ =
 
 (* Liboject entries of declared ML Modules *)
 
+(* Digest of module used to compile the file *)
+type ml_module_digest =
+  | NoDigest
+  | AnyDigest of Digest.t             (* digest of any used cma / cmxa *)
+
 type ml_module_object = {
   mlocal : Vernacexpr.locality_flag;
-  mnames : string list
+  mnames : (string * ml_module_digest) list
 }
 
+let add_module_digest m =
+  try
+    let file = file_of_name m in
+    let path, file = System.where_in_path ~warn:false !coq_mlpath_copy file in
+    m, AnyDigest (Digest.file file)
+  with
+  | Not_found ->
```

Comment:
In what kind of scenarios do we expect the file to be missing, but `trigger_ml_object` to succeed?

Reply:
A wrong `Declare Module` for example; this shouldn't produce not found. As I mentioned in the pull request we could refactor this to be better but I'll involve some more cleanups.

Also, there is the scenario that could be controlled by a flag to request the digests not to be stored; for whatever the reasons. Hard to know yet what kind of impact this has.

Reply:
Hi, I fell on a situation where `add_module_digest` fails, it is when configured with `-natdynlink no`. I'm not about what is the right fix but the following e.g. works:
```
let add_module_digest m =                                                              
  if module_is_known m then                                                            
    m, NoDigest                                                                        
  else                                                                                 
  try                                                                                  
    let file = file_of_name m in                                                       
    let path, file = System.where_in_path ~warn:false !coq_mlpath_copy file in         
    m, AnyDigest (Digest.file file)                                                    
  with                                                                                 
  | Not_found ->                                                                       
    m, NoDigest   
```